### PR TITLE
Multi-read supported chardev.c without race condition of offset.

### PR DESCRIPTION
**Pull request by YLowy**

### CODE REVIEW

File: examples/chardev.c [hunk 1]
```diff
@@ -13,6 +13,13 @@
 #include <linux/module.h>
 #include <linux/poll.h>
 
+/* Structure for every task's offset */
+typedef struct element{
+    int pid;
+    long offset;
+    struct element *next;
+}list_ele_t;
```
Comment (jserv) [inline examples/chardev.c:21]: Run `clang-format -i chardev.c` in directory `examples`.

**Review by jserv — CHANGES_REQUESTED** [review]
Follow Linux kernel coding style and avoid using camelCase such as `currentOffset`.

**Comment by linD026** [comment]
This improvement with the multi-read is unnecessary because the read operation of the `struct file` already being atomic since Linux v3.14.
You can see more [here](https://github.com/sysprog21/lkmpg/pull/108#issuecomment-923965811).
Ps. Apologize for the previously misleading information that I talked about in the private messages.

**Comment by jserv** [comment]
Close as per @linD026 suggested.

**Pull request closed without merging**